internal/vds/vds_test.go

```diff
@@ -425,6 +425,112 @@ func TestFence(t *testing.T) {
 	}
 }
 
+func TestFenceNearestInterpolationSnap(t *testing.T) {
+	testcases := []struct {
+		coordinate_system int
+		coordinates       [][]float32
+		expected          []float32
+	}{
+		{
+			coordinate_system: CoordinateSystemAnnotation,
```

Comment:
There should be a case for `x.4999`, together with the `x.5`. As we expect the first to round to `x` and the second to `x + 1`. This boundary is more interesting to test than the `x.25` and `x.75` cases